drivers/AnalogIn.h

```diff
@@ -93,6 +95,35 @@ class AnalogIn {
      */
     unsigned short read_u16();
 
+    /**
+     * Read the input voltage in volts. The output depends on the target board's
+     * ADC reference voltage (typically equal to supply voltage). The ADC reference voltage
+     * sets the maximum voltage the ADC can quantify (ie: Vin == Vref when ADC output == ADC_MAX_VALUE)
+     *
+     * The target's ADC reference voltage can be configured by overriding "target.adc_vref"
+     *
+     * @returns A floating-point value representing the current input voltage, measured in volts.
+     */
+    float read_voltage();
+
+    /**
+     * Sets this AnalogIn instance's reference voltage.
+     *
+     * Defaults to the configurable MBED_CONF_TARGET_DEFAULT_ADC_VREF setting.
```

Comment:
This isn't the place for that - by the time you're considering this call the default was determined by the constructor, and may not be the config setting. Discuss the config option in the constructor docs (in the missing `vref` parameter), and just refer to the constructor here.

And elsewhere you talk of "target.default_adc_vref`. Better to use that form than the define name.